vision/realsense/realsense.py

```diff
@@ -0,0 +1,143 @@
+        Resolution width of the pre-recorded .bag file
+    screen_height: number
+        Resolution height of the pre-recorded .bag file
+    frame_rate: number
+        Framerate of the pre-recorded .bag file
+    serial_no: str
+        Serial number of the realsense camera to stream from
+        Defaults to empty, which reads if only one realsense is plugged in
+    """
+    def __init__(self, screen_width, screen_height, frame_rate, serial_no=""):
+        super().__init__(screen_width, screen_height, frame_rate)
+
+        # Initialize clipping constants
+        self.CLIPPING = False
+        self.clipping_distance_in_meters = 4
+
+        self.serialNumber = serial_no
+
+        self.pipeline = rs.pipeline()
+
+        # Create a config object
+        self.config = rs.config()
+
+        # a unique serial number for a camera, enables config to get data from a specific camera
+        if self.serialNumber:
+            self.config.enable_device(self.serialNumber)
+
+        self.config.enable_stream(rs.stream.depth, self.width, self.height, rs.format.z16, self.framerate)
+        self.config.enable_stream(rs.stream.color, self.width, self.height, rs.format.bgr8, self.framerate)
+
+
+    def __iter__(self):
+        """
+        Iterates through each depth/color frame in the object
+
+        Parameters
+        ------------
+        self: Realsense object
+            automatically passed, you do not need to input it as an argument
+
+        Raises
+        ------------
+        NotImplementedError: if called on a base camera object
+
+        Returns
+        -----------
+        depth image: np array, 1 channel
+        color image: np array, 3 channels
+        """
+        # Start streaming from file
+        profile = self.pipeline.start(self.config)
+
+        # Getting the depth sensor's depth scale (see rs-align example for explanation)
+        depth_sensor = profile.get_device().first_depth_sensor()
+        depth_scale = depth_sensor.get_depth_scale()
+
+        # We will be removing the background of objects more than
+        #  clipping_distance_in_meters meters away
+        if self.CLIPPING:
+            clipping_distance = self.clipping_distance_in_meters / depth_scale
+
+        align_to = rs.stream.color
+        align = rs.align(align_to)
+
+        while True:
+            # returns the next color/depth frame
+            frames = self.pipeline.wait_for_frames()
+
+            # Align the depth frame to color frame
+            aligned_frames = align.process(frames)
+
+            # Get aligned frames
+            # aligned_depth_frame is a 640x480 depth image
+            aligned_depth_frame = aligned_frames.get_depth_frame()
+            color_frame = aligned_frames.get_color_frame()
+
+            depth_image = np.asanyarray(aligned_depth_frame.get_data())
+            color_image = np.asanyarray(color_frame.get_data())
+
+            yield depth_image, color_image
+
+    def display_in_window(self):
```

Comment:
They have enough distinctions where it might become a mess of if statements to check if a certain line should be run, and with the addition of the sim camera things would get even messier. I could make a function in the base Camera class that does the first, nearly identical half of the function, which I can then call from each display_in_window, though, if that would work better.

Reply:
Okay that is fine then.